src/GafferImage/ImageAlgo.cpp

```diff
@@ -118,6 +119,96 @@ int tileIndex( const Imath::V2i &tileOrigin, const Imath::Box2i &tileRange )
 	return offset.y * ( tileRange.size().x + 1 ) + offset.x;
 }
 
+// A comparison functor that tokenizes the string apart into numeric and non-numeric parts, and compares
+// the numeric parts according to the order of integers ( ie.  "x100x" > "x9x" )
+struct NaturalOrder
+{
+	bool operator()( const std::string &l, const std::string &r )
+	{
+		int moreZeros = 0;
+		std::sregex_iterator lit( l.begin(), l.end(), g_naturalSortRegex );
+		std::sregex_iterator rit( r.begin(), r.end(), g_naturalSortRegex );
+		std::sregex_iterator end;
+		for( ; lit != end && rit != end; ++lit, ++rit )
+		{
+			int c = compareToken( *lit, *rit );
+			if( c != 0 )
+			{
+				// We've found a difference in this set of tokens, and can return
+				return c < 0;
+			}
+
+			if( lit->str().size() != rit->str().size() )
+			{
+				// This is a weird special case - the strings have compared equal, but their lengths aren't
+				// the same.  This must be due to one having numbers with different zero padding.  If we
+				// find a significant difference, we'll go by that ( ie "0B" comes after "000A" ), but if
+				// we get to the end without finding a difference, we'll sort the string with more padding
+				// after, just to make sure we do something consistent.
+				moreZeros = lit->str().size() > rit->str().size() ? 1 : -1;
+			}
+		}
+
+		if( lit != end || rit != end )
+		{
+			// One of the strings still has tokens left, the string with more tokens is greater
+			return rit != end;
+		}
+
+		if( moreZeros != 0 )
+		{
+			return moreZeros < 0;
+		}
+
+		return false;
+	}
+
+private:
+
+	int compareToken( const std::smatch &l, const std::smatch &r )
+	{
+		// We only need to do anything special if both tokens are a number - otherwise, the standard
+		// string compare will just put the number before the letters
+		if ( !l[ 1 ].str().empty() && !r[ 1 ].str().empty() )
+		{
+			// We also only need to do something special if the lengths don't match - if the lengths
+			// match, then string comparison is equivalent to numerical comparison
+			if( l.str().size() < r.str().size() )
+			{
+				return -compareLongerNumber( r.str(), l.str() );
+			}
+			else if( l.str().size() > r.str().size() )
+			{
+				return compareLongerNumber( l.str(), r.str() );
+			}
+		}
+
+		return l.str().compare( r.str() );
+	}
+
+	int compareLongerNumber( const std::string &longer, const std::string &shorter )
+	{
+		int diff = longer.size() - shorter.size();
+		// If the longer string has any non-zero digits in the longer portion, it is greater.
+		for( int i = 0; i < diff; i++ )
+		{
+			if( longer[i] != '0' )
+			{
+				return 1;
+			}
+		}
+
+		// If the mismatched portion is all zeros, ignore it, and compare the matching portion
+		return longer.compare( diff, string::npos, shorter );
+	}
+
+	static const std::regex g_naturalSortRegex;
+
+};
+
+const std::regex NaturalOrder::g_naturalSortRegex( "(\\d+)|([^\\d]+)" );
```

Comment:
Might be nice to use raw string literals here, to avoid the escaping of `\`? See https://github.com/GafferHQ/gaffer/blob/main/src/GafferSceneUI/OutputBuffer.cpp#L103 for an example.